### ISSUE TITLE
A question about MinCutPool implementation

### ISSUE THREAD

**Issue by kaize0409** [quqixun/GNN-Pytorch#1]
Hi Qixun,

Thanks for your great efforts in reimplementing all those models! I have a simple question about the implementation details of MincutPool:

For layers.py (184 - 188) in MinCutPool, we are supposed to get the degree matrix of A, but I feel a bit lost about your implementation, could you provide some clarifications about that?

---------------------------------

D_flat = torch.sum(A, dim=-1)
D_eye = torch.eye(D_flat.size(1)).type_as(A).to(device)
D_flat = D_flat.unsqueeze(2).expand(*D_flat.size(), D_flat.size(1))
D = D_eye * D_flat

**Comment by quqixun** [comment]
Thanks for your attention.

I just copyed the implementation of torch_geometric.nn.dense_mincut_pool.

In this part, my implementation was commented as line 185 to get the degree matrix for each sample in A.
In my experiments, line 185 did same thing as line 186~188 in forward phase.
However, inconsistent output occured while in backward phase.
I did not fixed it out.

```python
184    D_flat = torch.sum(A, dim=-1)
185    # D = torch.cat([torch.diag(d).unsqueeze(0) for d in D_flat]).to(device)
186    D_eye = torch.eye(D_flat.size(1)).type_as(A).to(device)
187    D_flat = D_flat.unsqueeze(2).expand(*D_flat.size(), D_flat.size(1))
188    D = D_eye * D_flat
```

You can check output of line 186~188 step by step.
